app/controllers/submit.js

```diff
@@ -73,6 +73,17 @@ function subjectIdMap(subjectArray) {
     return subjectArray.map(subjectBlock => subjectBlock.map(subject => subject.id));
 }
 
+function doiRegexExec(doi) {
+    //Strips url out of inputted doi, if any.  For example, user input this DOI: https://dx.doi.org/10.12345/hello. Returns 10.12345/hello.
+    const doiRegex = /\b(10\.\d{4,}(?:\.\d+)*\/\S+(?:(?!["&\'<>])\S))\b/;
```

Comment:
Used your idea, to replace the input when the box loses focus.  I fire it `onchange`. Still using `regexObj.exec(str)` because of the different combinations the url could take (http, https..).  However, if there is no match, I don't change the errored DOI, I leave as is.  If there is a match, I replace the input.